File: WebApi/Interfaces/ITrSBoxService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrSBoxDto, CreateTrSBoxDto, UpdateTrSBoxDto } from '../Models/TrSBoxDtos';

export interface ITrSBoxService {
  getAll(): Promise<ApiResponse<TrSBoxDto[]>>;
  getById(id: number): Promise<ApiResponse<TrSBoxDto>>;
  getByImportLineId(importLineId: number): Promise<ApiResponse<TrSBoxDto[]>>;
  getByBoxId(boxId: number): Promise<ApiResponse<TrSBoxDto[]>>;
  getBySerialNumber(serialNumber: string): Promise<ApiResponse<TrSBoxDto[]>>;
  getBySerialNo(serialNo: string): Promise<ApiResponse<TrSBoxDto[]>>;
  getActive(): Promise<ApiResponse<TrSBoxDto[]>>;
  getByDescription(description: string): Promise<ApiResponse<TrSBoxDto[]>>;
  searchByDescription(description: string): Promise<ApiResponse<TrSBoxDto[]>>;
  create(createDto: CreateTrSBoxDto): Promise<ApiResponse<TrSBoxDto>>;
  update(id: number, updateDto: UpdateTrSBoxDto): Promise<ApiResponse<TrSBoxDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}

File: Interfaces/ITrSBoxService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrSBoxService
    {
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetAllAsync();
        Task<ApiResponse<TrSBoxDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetByImportLineIdAsync(long importLineId);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetByBoxIdAsync(long boxId);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetBySerialNumberAsync(string serialNumber);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetBySerialNoAsync(string serialNo);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> GetByDescriptionAsync(string description);
        Task<ApiResponse<IEnumerable<TrSBoxDto>>> SearchByDescriptionAsync(string description);
        Task<ApiResponse<TrSBoxDto>> CreateAsync(CreateTrSBoxDto createDto);
        Task<ApiResponse<TrSBoxDto>> UpdateAsync(long id, UpdateTrSBoxDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}